samcli/hook_packages/terraform/hooks/prepare/translate.py

```diff
@@ -369,11 +404,75 @@ def _link_lambda_functions_to_layers(
         source_resource_cfn_resource=lambda_funcs_conf_cfn_resources,
         source_resource_tf_config=lambda_config_funcs_conf_cfn_resources,
         destination_resource_tf=lambda_layers_terraform_resources,
-        intrinsic_type=LinkerIntrinsics.Ref,
-        cfn_intrinsic_attribute=None,
-        source_link_field_name="Layers",
+        tf_destination_attribute_name="arn",
         terraform_link_field_name="layers",
+        cfn_link_field_name="Layers",
         terraform_resource_type_prefix=LAMBDA_LAYER_RESOURCE_ADDRESS_PREFIX,
+        cfn_resource_update_call_back_function=_link_lambda_functions_to_layers_call_back,
+        linking_exceptions=exceptions,
+    )
+    ResourceLinker(resource_linking_pair).link_resources()
+
+
+def _link_gateway_methods_to_gateway_rest_apis_call_back(
+    gateway_method_cfn_resource: Dict, referenced_rest_apis_values: List[ReferenceType]
+) -> None:
+    """
+    Callback function that used by the linking algorithm to update an Api Gateway Method CFN Resource with
+    a reference to the Rest Api resource.
+
+    Parameters
+    ----------
+    gateway_method_cfn_resource: Dict
+        API Gateway Method CFN resource
+    referenced_rest_apis_values: List[ReferenceType]
+        List of referenced REST API either as the logical id of REST API resource defined in the customer project, or
+        ARN values for actual REST API resource defined in customer's account. This list should always contain one
+        element only.
+    """
+    # if the destination rest api list contains more than one element, so we have an issue in our linking logic
+    if len(referenced_rest_apis_values) > 1:
+        raise InvalidResourceLinkingException("Could not link multiple Rest APIs to one Gateway method resource")
+
+    ref_list = [
+        {"Ref": logical_id.value} if isinstance(logical_id, LogicalIdReference) else logical_id.value
+        for logical_id in referenced_rest_apis_values
+    ]
+    gateway_method_cfn_resource["Properties"]["RestApiId"] = ref_list[0]
```

Comment:
nit: maybe slightly more readable since we already know there is only one item in the list
```suggestion
    logical_id = referenced_rest_apis_values[0]
    gateway_method_cfn_resource["Properties"]["RestApiId"] = {"Ref": logical_id.value} if isinstance(logical_id, LogicalIdReference) else logical_id.value
```

Reply:
done